### ISSUE TITLE
Add support for PermissionModel and AutoDeployment attributes in aws_cloudformation_stack_set resource

### ISSUE THREAD

**Issue by dkpiesak** [hashicorp/terraform-provider-aws#14454]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, the `aws_cloudformation_stack_set` resource does not support attributes that pass the `PermissionModel` and `AutoDeployment` properties of a CloudFormation StackSet. These two properties are quite essential as they enable auto-deployment whenever the template of a StackSet changes.

### New or Affected Resource(s)

* aws_cloudformation_stack_set

### Potential Terraform Configuration

```hcl
resource "aws_cloudformation_stack_set" "example" {
  administration_role_arn = aws_iam_role.AWSCloudFormationStackSetAdministrationRole.arn
  name                    = "example"

  parameters = {
    VPCCidr = "10.0.0.0/16"
  }

  template_body = <<TEMPLATE>>

  permission_model = "SERVICE_MANAGED" | "SELF_MANAGED"
  auto_deployment {
    enabled = True | False
    retain_stacks_on_account_removal = True | False
  }
}
```

### References

* https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateStackSet.html
* https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack-set.html
* https://docs.aws.amazon.com/cli/latest/reference/cloudformation/update-stack-set.html
* https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_UpdateStackSet.html

**Comment by srikanthchelluri** [comment]
I believe this is a duplicate of https://github.com/terraform-providers/terraform-provider-aws/issues/12422.

**Comment by bflad** [comment]
Indeed, this is a duplicate of #12422, which has just been merged and will release with version 3.35.0 of the Terraform AWS Provider. 👍
